app/src/main/java/com/example/campusguide/directions/ChooseDestinationOptions.kt

```diff
@@ -43,12 +48,15 @@ class ChooseDestinationOptions(private val locationSelectedListener: (location:
     }
 
     private fun useNextEvent(){
-        //TODO fill this method once we have the logic to use a calendar Event
         dismiss()
-        val location = Location("Montreal")
-        location.latitude = 45.5017
-        location.longitude = -73.5673
-        locationSelectedListener(location)
+        val act = activity as MapsActivity
+        val calendarBox: Box<Calendar> = ObjectBox.boxStore.boxFor()
+        val calendar = Pair(calendarBox.all[0].id, calendarBox.all[0].name)
+
+        val nextLocation = Events(act, calendar).getNextEventLocation()
+        GlobalScope.launch {
+            FindEventLocation(act, locationSelectedListener).getLocationOfEvent(nextLocation)
+        }
```

Comment:
Needs exception handling and to tell the user to log-in or select a calendar if they haven't already.

Reply:
The app crashes if user isn't logged in or hasn't selected a calendar.

Reply:
Something like the "onError" you did for when there are no events on a given day

Reply:
This is a known issue that will be logged later on and fixed in the next sprint.